api/urls.py

```diff
@@ -58,6 +58,9 @@ def generate_viewset(self, subclass, serializer):
         if subclass.__name__ == 'AgentIdentifier':
             queryset = queryset.exclude(scheme='mailto')
 
+        if 'is_deleted' in [field.name for field in serializer.Meta.model._meta.get_fields()]:
```

Comment:
Instead of iterating through all the fields, I think it would be cleaner and faster to try `get_field('is_deleted')` and catch `FieldDoesNotExist`